octree/include/pcl/octree/impl/octree_iterator.hpp

```diff
@@ -133,15 +133,16 @@ namespace pcl
         if ( (this->max_octree_depth_>=stack_entry.depth_) &&
              (stack_entry.node_->getNodeType () == BRANCH_NODE) )
         {
-          unsigned char child_idx;
-
           // current node is a branch node
           BranchNode* current_branch =
               static_cast<BranchNode*> (stack_entry.node_);
 
+          unsigned char child_idx = 8;
+
           // add all children to stack
-          for (child_idx = 0; child_idx < 8; ++child_idx)
+          do
```

Comment:
> Also I would like to keep child_idx as a unsigned char because if I change it, I get a warning on line where a method expect a unsigned char as argument, if you see what I mean.

I like for-loops

```cpp
for (int8_t i = 7; i >= 0; --i)
{
  const unsigned char child_idx = i;
  [...]
}
```

Reply:
I'm OK with that.